public/src/overmind/namespaces/review/actions.ts

```diff
@@ -23,7 +23,11 @@ export const updateReview = async ({ state, actions, effects }: Context): Promis
         const response = await effects.grpcMan.updateReview(review, state.activeCourse)
         if (success(response) && response.data) {
             // Updates the currently selected review with the new data from the server
-            state.review.reviews[state.activeCourse.toString()][state.activeSubmission][state.review.selectedReview] = response.data
+            const reviews = new Map(state.review.reviews)
+            reviews.set(state.activeSubmission, state.review.reviews.get(state.activeSubmission)?.map(r => r.ID === review.ID ? response.data as Review : r) ?? [])
```

Comment:
Too long line. Use more lines or make a method if this would be of use in other places.